pkg/osvscanner/scan.go

```diff
@@ -7,19 +7,66 @@ import (
 	"path/filepath"
 
 	"github.com/google/osv-scalibr/extractor"
+	"github.com/google/osv-scalibr/extractor/filesystem"
+	"github.com/google/osv-scalibr/extractor/filesystem/language/java/pomxmlnet"
 	"github.com/google/osv-scanner/v2/internal/imodels"
 	"github.com/google/osv-scanner/v2/internal/scalibrextract"
 	"github.com/google/osv-scanner/v2/internal/scalibrextract/ecosystemmock"
+	"github.com/google/osv-scanner/v2/internal/scalibrextract/filesystem/vendored"
+	"github.com/google/osv-scanner/v2/internal/scalibrextract/language/java/pomxmlenhanceable"
+	"github.com/google/osv-scanner/v2/internal/scalibrextract/vcs/gitrepo"
 	"github.com/google/osv-scanner/v2/pkg/osvscanner/internal/scanners"
+	"github.com/ossf/osv-schema/bindings/go/osvschema"
 )
 
+func configureExtractors(extractors []filesystem.Extractor, accessors ExternalAccessors, actions ScannerActions) {
+	for _, tor := range extractors {
+		if accessors.DependencyClients[osvschema.EcosystemMaven] != nil && accessors.MavenRegistryAPIClient != nil {
+			pomxmlenhanceable.EnhanceIfPossible(tor, pomxmlnet.Config{
+				DependencyClient:       accessors.DependencyClients[osvschema.EcosystemMaven],
+				MavenRegistryAPIClient: accessors.MavenRegistryAPIClient,
+			})
+		}
+
+		// todo: the "disabled" aspect should probably be worked into the extractor being present in the first place
+		//  since "IncludeRootGit" is always true
+		gitrepo.Configure(tor, gitrepo.Config{
+			IncludeRootGit: actions.IncludeGitRoot,
+			Disabled:       !actions.IncludeGitRoot,
+		})
+
+		vendored.Configure(tor, vendored.Config{
+			// Only attempt to vendor check git directories if we are not skipping scanning root git directories
+			ScanGitDir: !actions.IncludeGitRoot,
+			OSVClient:  accessors.OSVDevClient,
+			Disabled:   accessors.OSVDevClient == nil,
+		})
+	}
+}
+
+func getExtractors(defaultExtractorNames [][]string, accessors ExternalAccessors, actions ScannerActions) []filesystem.Extractor {
+	names := actions.ExtractorNames
+
+	if len(names) == 0 {
+		for _, preset := range defaultExtractorNames {
```

Comment:
nit: slices.Concat can be used here. I think that means we can get rid of the [][]string and just have a single [] and have the caller call slices.Concat themselves.